kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/DeploymentDescriptorServiceCDI.java

```diff
@@ -0,0 +1,89 @@
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.guvnor.m2repo.backend.server.GuvnorM2Repository;
+import org.jboss.errai.bus.server.annotations.Service;
+import org.kie.internal.runtime.conf.RuntimeStrategy;
+import org.kie.internal.runtime.manager.deploy.DeploymentDescriptorIO;
+import org.kie.workbench.common.screens.server.management.model.ProcessConfigModule;
+import org.kie.workbench.common.screens.server.management.service.DeploymentDescriptorService;
+import org.kie.workbench.common.services.backend.kmodule.KModuleContentHandler;
+import org.kie.workbench.common.services.shared.kmodule.KBaseModel;
+import org.kie.workbench.common.services.shared.kmodule.KSessionModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import static org.kie.workbench.common.screens.server.management.model.RuntimeStrategy.*;
+
+@Service
+@ApplicationScoped
+public class DeploymentDescriptorServiceCDI implements DeploymentDescriptorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeploymentDescriptorServiceCDI.class);
+
+    private GuvnorM2Repository guvnorM2Repository;
+
+    @Inject
+    public DeploymentDescriptorServiceCDI(GuvnorM2Repository guvnorM2Repository) {
+        this.guvnorM2Repository = guvnorM2Repository;
+    }
+
+    @Override
+    public ProcessConfigModule getProcessConfig(String path) throws IOException {
+        ProcessConfigModule processConfig = new ProcessConfigModule();
+        KModuleContentHandler kModuleContentHandler = new KModuleContentHandler();
+        try (InputStream in = IOUtils.toInputStream(guvnorM2Repository.getKieDeploymentDescriptorText(path), StandardCharsets.UTF_8);) {
+            RuntimeStrategy runtimeStrategy = DeploymentDescriptorIO.fromXml(in).getRuntimeStrategy();
+
+            switch (runtimeStrategy) {
+                case SINGLETON:
+                    processConfig.setRuntimeStrategy(SINGLETON);
+                    break;
+                case PER_CASE:
+                    processConfig.setRuntimeStrategy(PER_CASE);
+                    break;
+                case PER_PROCESS_INSTANCE:
+                    processConfig.setRuntimeStrategy(PER_PROCESS_INSTANCE);
+                    break;
+                case PER_REQUEST:
+                    processConfig.setRuntimeStrategy(PER_REQUEST);
+                    break;
+            }
+        } catch (Exception e) {
+            LOGGER.warn(e.getMessage());
+        }
+
+        Map<String, KBaseModel> kBaseModelHashMap = kModuleContentHandler.toModel(guvnorM2Repository.getKModuleText(path)).getKBases();
+        AtomicReference<KBaseModel> kBaseModelAtomicReference = new AtomicReference<>();
+        kBaseModelHashMap.forEach((s, kBaseModel) -> {
+            if (kBaseModel.isDefault()) {
+                kBaseModelAtomicReference.set(kBaseModel);
+            }
+        });
+
+        KBaseModel defaultKBaseModel = kBaseModelAtomicReference.get();
```

Comment:
Please avoid using AtomicReference.

```suggestion
        Optional<KBaseModel> defaultKBaseModel = kBaseModelHashMap.values().stream().filter(kBaseModel -> kBaseModel.isDefault()).findFirst();
        if (defaultKBaseModel.isPresent()) {
```